src/kudu/consensus/log_cache.cc

```diff
@@ -315,7 +315,16 @@ Status LogCache::ReadOps(int64_t after_op_index,
                          std::vector<ReplicateRefPtr>* messages,
                          OpId* preceding_op) {
   DCHECK_GE(after_op_index, 0);
-  RETURN_NOT_OK(LookupOpId(after_op_index, preceding_op));
+
```

Comment:
What is the expectation from caller side?
Could we be in a situation where preceding_opid is not populated since we failed to 
do LookupOpId and ReadReplicatesInRange succeeded, or is ReadReplicatesInRange going
to do the same LookupOpId again and fail but with a better error message/error manager handling?

Reply:
Yes, you were right. This introduced a slight behavior change. To retain the old behavior, I have made some changes.If LookupOpId() fails with NotFound(), then we try to read a single entry from the log - which gives it a chance to update the error manager to be consumed by upper layers.